src/app/components/mayastor/mayastor.component.ts

```diff
@@ -11,15 +13,28 @@ import { Status } from 'src/app/model/enum.model';
 })
 export class MayastorComponent implements OnInit {
   data: any;
-  constructor(private ApiService: DashboardData, private router: Router) {
-    ApiService.getMayastorTest().subscribe((res) => {
+  mySubscription: any;
+
+  constructor(private ApiService: DashboardData, private router: Router, private activatedRoute: ActivatedRoute) {
+    this.router.routeReuseStrategy.shouldReuseRoute = function () {
+      return false;
```

Comment:
What is the use/intention  here ?

Reply:
/mayastor/{ID} this one will update the data on each route change .
If this one wont present , route will change and component data not updating .